### PR TITLE
rotate passwords for AVS 2020-07-17-preview

### PR DESCRIPTION
**Pull request by ctaggart**
This removes a breaking change in the 2020-07-17-preview to make it compatible with the 2020-03-20 API. I'm backporting it from the 2021-01-01-preview API [in development](https://github.com/Azure/azure-rest-api-specs-pr/pull/2634). We are aware that this is a breaking change to the 2020-07-17-preview API and I will review it with the review board.  No SDKs have been published for this API version.  Rotate password functionality is not currently working in production. This will allow it to work.

This pull requests adds two actions for rotate passwords. Here is how the [actions can be invoked](https://docs.microsoft.com/en-us/cli/azure/resource?view=azure-cli-latest#az_resource_invoke_action):

```
az resource invoke-action --action rotateVcenterPassword  \
--ids /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}

az resource invoke-action --action rotateNsxtPassword \
--ids /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AVS/privateClouds/{privateCloudName}
```

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [x] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [ ] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/vmware/resource-manager/Microsoft.AVS/preview/2020-07-17-preview/vmware.json
Comment (filizt) [inline specification/vmware/resource-manager/Microsoft.AVS/preview/2020-07-17-preview/vmware.json:609]: Is the long running operation return any successful completion status at the end? If the action does not require a response you can use 204 NoContent. Find more here: [https://github.com/Azure/azure-resource-manager-rpc/blob/423d048e58ce13cb8863fb2b2b22b4d6057c5318/v1.0/Addendum.md#call-action-post-asynchronously](https://github.com/Azure/azure-resource-manager-rpc/blob/423d048e58ce13cb8863fb2b2b22b4d6057c5318/v1.0/Addendum.md#call-action-post-asynchronously)

**Comment by ctaggart** [comment]
@filizt, thanks for the feedback! I added the 204 response in 0d2b704.  I'd like to get ARM signoff on this, this week. I will do the internal breaking change meeting on Monday (internal [work item 9068467](https://msazure.visualstudio.com/One/_workitems/edit/9068467)).

**Comment by filizt** [comment]
@ctaggart The changes in the PR looks good from ARM perspective but official sign off will wait until breaking changes board's approval.

**Review by JeffreyRichter — APPROVED** [review]

**Comment by ctaggart** [comment]
@filizt & @zhenglaizhang, the breaking change has been reviewed and approved. Please approve & merge.

**Comment by jorgecotillo** [comment]
ARM Signed off

**Comment by zhenglaizhang** [comment]
@msyyc @ArcturusZhang pls help take a look at python or go sdk breaking changes respectively

**Comment by ctaggart** [comment]
@zhenglaizhang, we do not have any SDKs published for this API version. We do not intend to publish SDKs for this API version. We do need this merged soon.

**Review by ArcturusZhang — CHANGES_REQUESTED** [review]
Hi @ctaggart unfortunately we have released one package of this api-version in go SDK v49.0.0 (https://github.com/Azure/azure-sdk-for-go/tree/v49.0.0/services/preview/avs/mgmt/2020-07-17-preview/avs).
If we did not want to release one package, we should not include this package in `readme.go.md` (this was added in this PR https://github.com/Azure/azure-rest-api-specs/pull/10866). Could you please remove it from `readme.go.md`? 
Considering that this package is a preview package, it is OK to remove it. But I suggest you double confirm about this.

**Comment by ctaggart** [comment]
Hey @ArcturusZhang, in that case, I'm okay with releasing the go SDK with these changes.

**Review by ArcturusZhang — APPROVED** [review]
The breaking changes introduced in this PR are relative minor and considering those happen on a preview package, therefore those are acceptable

**Comment by msyyc** [comment]
> @msyyc Yuchao Yan FTE @ArcturusZhang Dapeng Zhang FTE pls help take a look at python or go sdk breaking changes respectively

It's ok for Python

**Pull request merged**
